### PR TITLE
typespec, rename cadl-java emitter to typespec-java

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Link #2017 
This pr includes:
1. renames cadl-java emitter to typespec-java, update references in package.json
2. update pre-release-ci.yml
3. update tspconfig.yaml
4. update readme.md

### CODE REVIEW

File: cadl-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -928,7 +928,7 @@ export class CodeModelBuilder {
               if (match) {
                 schema = candidateResponseSchema;
                 this.program.trace(
-                  "cadl-java",
+                  "typespec-java",
                   `Replace Cadl model ${this.getName(bodyType)} with ${candidateResponseSchema.language.default.name}`,
```
Comment (weidongxu-microsoft) [inline cadl-extension/src/code-model-builder.ts:932]:
```suggestion
                  `Replace TypeSpec model ${this.getName(bodyType)} with ${candidateResponseSchema.language.default.name}`,
```

File: cadl-tests/package.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "@azure-tools/typespec-autorest": "~0.27.0",
     "@azure-tools/cadl-ranch-specs": "~0.8.0",
     "@azure-tools/cadl-ranch-expect": "~0.1.16",
-    "@azure-tools/cadl-java": "file:/../cadl-extension/azure-tools-cadl-java-0.4.2.tgz"
+    "@azure-tools/typespec-java": "file:/../cadl-extension/azure-tools-typespec-java-0.4.2.tgz"
```
Comment (weidongxu-microsoft) [inline cadl-tests/package.json:15]: 0.5.0?

**Comment by haolingdong-msft** [comment]
I've tested MVAD generation locally, it passes.

**Pull request merged**